File: Solutions/523. Continuous Subarray Sum/523.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        HashMap<Integer, Integer> hm = new HashMap<>();
        hm.put(0, -1);
        int sum = 0;
        int rem = 0;
        for(int i = 0; i < nums.length; i++){
            sum += nums[i];
            rem = sum % k;
            if(!hm.containsKey(rem)){
                hm.put(rem, i);
            }else if(i - hm.get(rem) > 1){
                return true;
            }
        }
        return false;
    }
}

File: Solutions/523. Continuous Subarray Sum/523.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        hm = { 0: -1}
        sum = 0
        for i in range(len(nums)):
            sum += nums[i]
            rem = sum % k
            if rem not in hm :
                hm[rem] = i
            elif i - hm[rem] > 1:
                return True
        return False